Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/SlimeMediator.cs

```diff
@@ -87,9 +95,39 @@ public void OnDeath()
             Divide();
         }
 
+        public void OnPlayerClose()
+        {
+            StartChasing();
+        }
+
+        public void OnPlayerFar()
+        {
+            StartPatrolling();
+        }
+
+        public void StartChasing()
+        {
+            _enemyPatrolling.SetPatrolling(false);
+            _slimeMovement.StartChasing();
+        }
+
+        public void StartPatrolling()
+        {
+            _slimeMovement.StopChasing();
+            _enemyPatrolling.SetPatrolling(true);
+        }
         public void OnHit()
         {
             throw new NotImplementedException();
         }
```

Comment:
Is this ment to be called when this enemy takes damage?

Reply:
Then replace the throw with a comment along the lines of "//In this case, do nothing".
If we leave exception throws it seems like we forgot to implement whatever was supposed to go there, so in these cases I would add a comment instead.